components/patina_samples/src/smbios_platform.rs

```diff
@@ -0,0 +1,342 @@
+                String::from("ASSET-MB-001"),
+                String::from("Main Board Slot"),
+            ],
+        };
+
+        let handle = smbios.add_record(None, &baseboard_info).map_err(|e| {
+            log::error!("Failed to add baseboard info: {:?}", e);
+            patina::error::EfiError::DeviceError
+        })?;
+
+        log::info!("  Type 2 (Baseboard Info) - Handle 0x{:04X}", handle);
+        Ok(())
+    }
+
+    /// Add Type 3 (System Enclosure/Chassis)
+    ///
+    /// Demonstrates adding chassis type, manufacturer, and serial information.
+    fn add_system_enclosure(smbios: &Service<Smbios>) -> Result<()> {
+        let enclosure_info = Type3SystemEnclosure {
+            header: SmbiosTableHeader::new(3, 0, SMBIOS_HANDLE_PI_RESERVED),
+            manufacturer: 1,
+            enclosure_type: 0x03, // Desktop
+            version: 2,
+            serial_number: 3,
+            asset_tag_number: 4,
+            bootup_state: 0x03,
+            power_supply_state: 0x03,
+            thermal_state: 0x03,
+            security_status: 0x02,
+            oem_defined: 0x00000000,
+            height: 0x00,
+            number_of_power_cords: 0x01,
+            contained_element_count: 0x00,
+            contained_element_record_length: 0x00,
+            string_pool: vec![
+                String::from("Example Corporation"),
+                String::from("Example Chassis v1.0"),
+                String::from("CHASSIS-99999"),
+                String::from("ASSET-CHASSIS-001"),
+            ],
+        };
+
+        let handle = smbios.add_record(None, &enclosure_info).map_err(|e| {
+            log::error!("Failed to add system enclosure: {:?}", e);
+            patina::error::EfiError::DeviceError
+        })?;
+
+        log::info!("  Type 3 (System Enclosure) - Handle 0x{:04X}", handle);
+        Ok(())
+    }
+
+    /// Add custom vendor-specific OEM record (Type 0x80)
+    ///
+    /// Demonstrates how platforms can add custom vendor-specific SMBIOS records.
+    /// Types 0x80-0xFF are reserved for OEM/vendor use.
+    fn add_vendor_oem_record(smbios: &Service<Smbios>) -> Result<()> {
+        let vendor_record = VendorOemRecord {
+            header: SmbiosTableHeader::new(VendorOemRecord::RECORD_TYPE, 0, SMBIOS_HANDLE_PI_RESERVED),
+            oem_field: 0xDEADBEEF, // Example vendor-specific data
+            string_pool: vec![String::from("Example Vendor"), String::from("Platform Data v1.0")],
+        };
+
+        let handle = smbios.add_record(None, &vendor_record).map_err(|e| {
+            log::error!("Failed to add vendor OEM record: {:?}", e);
+            patina::error::EfiError::DeviceError
+        })?;
+
+        log::info!("  Type 0x80 (Vendor OEM) - Handle 0x{:04X}", handle);
+        Ok(())
+    }
+
+    /// Add Type 127 (End of Table) marker
+    ///
+    /// Required marker to indicate the end of the SMBIOS table.
+    fn add_end_of_table(smbios: &Service<Smbios>) -> Result<()> {
```

Comment:
Not a bug, more of a question on implementation.  The original SMBIOS driver in Tianocore automatically creates the end of table marker for the caller.

[SmbiosDxe.c](https://github.com/tianocore/edk2/blob/master/MdeModulePkg/Universal/SmbiosDxe/SmbiosDxe.c#L1142)

This allows the caller to add tables in any order at any time during boot without having to know they need to provide an end table marker.  Do we want to follow the same usage model?